### ISSUE TITLE
V8 12.7.224.18 build failure on Linux ARM64 GCC

### ISSUE THREAD

**Issue by tommie** [tommie/v8go#66]
https://github.com/tommie/v8go/actions/runs/10500092541/job/29088024717

```
FAILED: obj/third_party/zlib/zlib_arm_crc32/crc32_simd.o 
ccache aarch64-linux-gnu-gcc -MD -MF obj/third_party/zlib/zlib_arm_crc32/crc32_simd.o.d -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_OZONE=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNO_UNWIND_TABLES -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -D_GLIBCXX_ASSERTIONS=1 -D__ARM_NEON__=1 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DZLIB_IMPLEMENTATION -DCRC32_ARMV8_CRC32 -DARMV8_OS_LINUX -I../../v8/third_party/zlib -I../../v8 -Igen -Wall -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-deprecated-declarations -Wno-comments -Wno-packed-not-aligned -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -fno-ident -fno-strict-aliasing -fstack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -pipe -pthread -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -O2 -fdata-sections -ffunction-sections -fno-math-errno -fno-omit-frame-pointer -g0 -fvisibility=hidden -O3 -fdata-sections -ffunction-sections -fno-math-errno -march=armv8-a+aes+crc -std=gnu11 -c ../../v8/third_party/zlib/crc32_simd.c -o obj/third_party/zlib/zlib_arm_crc32/crc32_simd.o
{standard input}: Assembler messages:
{standard input}:151: Error: selected processor does not support `pmull v20.1q,v1.1d,v0.1d'
```

I think this is due to https://chromium-review.googlesource.com/c/chromium/src/+/5529831, fixed by https://chromium-review.googlesource.com/c/chromium/src/+/5646283 and https://chromium-review.googlesource.com/c/chromium/src/+/5746951.

* We are at e6286d70c55da94a8e12658918989b9b9dc325c8 ([12.7.224.18](https://github.com/v8/v8/tree/12.7.224.18)).
* Which points to Chromium zlib https://chromium.googlesource.com/chromium/src/third_party/zlib/+/209717dd69cd62f24cbacc4758261ae2dd78cfac/
* The zlib dependency has been rolled forward multiple times since the latest release.
* Looks like the Chromium dashboard version is way behind the latest V8 release (13.0.46).

Perhaps Chromium Linux arm64 isn't built using GCC, so that's why their builds are passing?

(I don't know why v8go is using GCC. It's carried over from the original.)

**Comment by tommie** [comment]
Looks like I've fixed the workflows so it doesn't leave a partial upgrade (`deps/v8_hash`) committed when builds are failing. Yay. So I think the easiest solution is to simply wait for Chromium to pick up a working V8.

**Comment by tommie** [comment]
Added ability to block upgrades to certain hashes: https://github.com/tommie/v8go/commit/efd2effe6313f17fc237362a7e6dc6fe8fd76cbe to stop v8upgrade from nagging.

I don't think we need to list all bad hashes, since the newer Chromium channels are already past the fixes:

* Beta: 4a42f512c8185ea9de592105ac0058c44f50b19c (zlib d3aea2341cdeaf7e717bc257a59aa7a9407d318a)
* Dev: 6b9601f192dc05725caef8f5d9513049ee124a1a (zlib d3aea2341cdeaf7e717bc257a59aa7a9407d318a)

https://chromiumdash.appspot.com/fetch_releases?platform=Linux&num=1&offset=0

**Comment by tommie** [comment]
It wasn't enough... https://github.com/tommie/v8go/actions/runs/10606974974/job/29398735341

`chromium/6629` is the lowest V8 that has the PMULL fix. Beta and Dev are now at 6669.

**Comment by tommie** [comment]
https://github.com/tommie/v8go/actions/runs/10913799761 built and released as 0.18.0.

zlib is at d3aea2341cdeaf7e717bc257a59aa7a9407d318a.
